src/engines/aurora/walkeleveval.cpp

```diff
@@ -0,0 +1,153 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Class to evaluate walkmesh elevation at given coordinates.
+ */
+
+#include "src/common/maths.h"
+#include "src/common/util.h"
+
+#include "src/engines/aurora/walkeleveval.h"
+
+namespace Engines {
+
+struct Point {
+	float x, y, z;
+};
+
+struct Ray {
+	Point p0, p1;
+};
+
+struct Triangle {
+	Point v0, v1, v2;
+};
+
+inline Point add(const Point &u, const Point &v) {
+	return Point { u.x + v.x,
+	               u.y + v.y,
+	               u.z + v.z };
```

Comment:
clang on the Travis CI VMs doesn't like that, as you can see here: https://travis-ci.org/xoreos/xoreos/jobs/375982631#L1954 . That's C++11 syntax anyway, right?